subsys/senss/include/senss_sensor_virtual.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSS_SENSOR_VIRTUAL_H_
+#define SENSS_SENSOR_VIRTUAL_H_
+
+#include <senss/senss_datatypes.h>
+#include "senss_sensor.h"
+
+/**
+ * @brief Virtual Sensor API
+ * @addtogroup senss_virtual
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct senss_sensor_virtual_api
+ * @brief Virtual sensor callback api
+ *
+ * A virtual sensor must register this callback API during sensor register.
+ */
+struct senss_sensor_virtual_api {
```

Comment:
Virtual sensors should be the same as physical sensors:

- This is missing setting/getting range. Assume for example an angle calculator virtual sensor implementation. Most logical would be to have 0-360 degree range. But what if an application has a physical layout that only needs 0-15 degree but with higher accuracy? They should still be able to set the range.
- This is missing setting/getting offset. Similarly, applications might want to be able to leverage configuring a bias. Maybe an orientation virtual sensor has drift and the application has custom logic to cancel it out?
- Virtual sensors might want a one off read sample as well. In the lid example maybe we just want an instantaneous angle calculation. The application would request the sample from the handler for the lid angle, the lid angle would then (in a blocking call) request the accel data from the 2 sources, then return the calculated data.

As per fifo and calibration, I don't know off the top of my head what use cases might come up, but the developers can always just set those to `NULL`.

Finally, `process` can simply be done the same as any other client of the sensor subsystem. There's no need for the sensor to get special treatment.

Reply:
Finally, you convinced me! :)

Separated APIs can help developers focus on virtual sensor or physical sensor development, maybe this is not so critical, especially on opensource projects. 

And, as you mentioned, combined APIs also have many benefits. 

Updated with combined single sensor APIs for virtual sensor and physical sensor. For the APIs the sensor don't support, developers need set them to `NULL`.